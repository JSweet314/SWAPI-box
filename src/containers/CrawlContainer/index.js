import React, {Component} from 'react';
import {fetchOpeningCrawl} from '../../apiCalls/fetchOpeningCrawl';
import './style.css';

export default class CrawlContainer extends Component {
  constructor() {
    super();
    this.state = {openingCrawl: '', title: '', releaseDate: ''};
  }

  componentDidMount = () => {
    this.findClosestOpeningCrawl();
  }

  findClosestOpeningCrawl = () => {
    const priorCrawl = localStorage.getItem('SWAPI-crawl');
    if (priorCrawl) {
      this.setState({...JSON.parse(priorCrawl)});
    } else {
      this.getOpeningCrawl();
    }
  }

  getOpeningCrawl = () => {
    const randomFilmNumber = Math.floor(Math.random() * 7) + 1;
    fetchOpeningCrawl(randomFilmNumber)
      .then(this.deployNewCrawl)
      .catch(error => alert(error));
  }

  deployNewCrawl = filmData => {
    const {openingCrawl, title, releaseDate} = filmData;
    this.setState({openingCrawl, title, releaseDate}, this.storeOpeningCrawl);
  };

  storeOpeningCrawl = () => 
    localStorage.setItem('SWAPI-crawl', JSON.stringify(this.state))

  render() {
    const {openingCrawl, title, releaseDate} = this.state;
    return (
      <aside className="scrolling-text">
        <div className="crawl">
          <pre>{openingCrawl}</pre>
          <p>{title}</p>
          <p>{releaseDate}</p>
        </div>
        <h3 className="instructions">Select a Category</h3>
      </aside>
    );
  }
}