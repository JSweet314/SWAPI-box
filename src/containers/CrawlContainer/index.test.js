import React from 'react';
import CrawlContainer from './index';
import { shallow } from 'enzyme';

describe('ScrollingText', () => {
  it('should match a snapshot', () => {
    const wrapper = shallow(
      <CrawlContainer />, 
      {disableLifecycleMethods: true}
    );
    expect(wrapper).toMatchSnapshot();
  });
});