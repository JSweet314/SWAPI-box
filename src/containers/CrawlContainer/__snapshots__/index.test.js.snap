// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollingText should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CrawlContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="crawl"
>
          <pre />
          <p />
          <p />
</div>,
        <h3
          className="instructions"
>
          Select a Category
</h3>,
      ],
      "className": "scrolling-text",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <pre />,
            <p />,
            <p />,
          ],
          "className": "crawl",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "pre",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select a Category",
          "className": "instructions",
        },
        "ref": null,
        "rendered": "Select a Category",
        "type": "h3",
      },
    ],
    "type": "aside",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="crawl"
>
            <pre />
            <p />
            <p />
</div>,
          <h3
            className="instructions"
>
            Select a Category
</h3>,
        ],
        "className": "scrolling-text",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <pre />,
              <p />,
              <p />,
            ],
            "className": "crawl",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "pre",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select a Category",
            "className": "instructions",
          },
          "ref": null,
          "rendered": "Select a Category",
          "type": "h3",
        },
      ],
      "type": "aside",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
